java/src/jmri/jmrit/logixng/tools/swing/WhereUsedFrame.java

```diff
@@ -0,0 +1,320 @@
+package jmri.jmrit.logixng.tools.swing;
+
+import java.awt.*;
+import java.awt.event.*;
+import java.io.File;
+import java.io.IOException;
+
+import javax.swing.*;
+
+import jmri.*;
+import jmri.jmrit.entryexit.EntryExitPairs;
+import jmri.jmrit.logix.OBlockManager;
+import jmri.jmrit.logix.WarrantManager;
+import jmri.jmrit.logixng.util.WhereUsed;
+import jmri.swing.NamedBeanComboBox;
+import jmri.util.FileUtil;
+import jmri.util.swing.JComboBoxUtil;
+import jmri.util.swing.JmriJOptionPane;
+
+/**
+ * Create a where used report based on the selected bean.  The selection combo box is
+ * based on the selected type.
+ * <P>
+ * This class is a copy of jmri.jmrit.whereused.WhereUsedFrame, but it only shows
+ * where a bean is used by LogixNG. On the other hand, it shows where in the LogixNG
+ * tree the bean is used.
+ *
+ * @author Dave Sand Copyright (C) 2020
+ * @author Dave Sand Copyright (C) 20203
```

Comment:
2023

Reply:
Thanks. My intention was to add my name, but I failed to do so. Fixed now.